### PR TITLE
Added SonarCompositeRuleId helper class

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Prep for part of #771

### CODE REVIEW

File: src/Core/SonarCompositeRuleId.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ */
+
+using System;
+
+namespace SonarLint.VisualStudio.Core
+{
+    public class SonarCompositeRuleId
+    {
+        /// <summary>
+        /// Attempts to parse a VS error code as a Sonar rule in the form "[repo key]:[rule key]"
```
Comment (duncanp-sonar) [inline src/Core/SonarCompositeRuleId.cs:28]: As part of changing the "disable rule" command I need both parts of the composite rule ID so I've created this helper.

File: src/Core/SonarCompositeRuleId.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        /// </summary>
+        public static bool TryParse(string vsErrorCode, out SonarCompositeRuleId ruleInfo)
```
Comment (rita-sonarsource) [inline src/Core/SonarCompositeRuleId.cs:30]: vsErrorCode is a bit misleading, it sounds like something VS would throw. Maybe `errorListRuleCode`?

**Review by rita-sonarsource — APPROVED** [review]

File: src/Core/SonarCompositeRuleId.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/Core/SonarCompositeRuleId.cs:30]: The column in the Error List is called "ErrorCode" - `ErrorListErrorCode`?

Comment (rita-sonarsource) [reply src/Core/SonarCompositeRuleId.cs:30]: Yeah that works 👍

**Pull request merged**